bots/elasticsearch/tasks.py

```diff
@@ -132,3 +136,73 @@ def serialize(self, source):
             'name': safe_substr(source.long_title),
             'short_name': safe_substr(source.name)
         }
+
+
+class JanitorTask(AppTask):
+    '''
+    Looks for discrepancies between postgres and elastic search numbers
+    Re-indexes time periods that differ in count
+    '''
+    def check_counts_in_range(self, min_date, max_date):
+        partial_database_count = CreativeWork.objects.exclude(title='').exclude(is_deleted=True).filter(
+            date_created__range=[min_date.isoformat(), max_date.isoformat()]
+        ).count()
+        partial_es_count = self.es_client.count(
+            index=(self.es_index or settings.ELASTICSEARCH_INDEX),
+            doc_type='creativeworks',
+            body={
+                'query': {
+                    'range': {
+                        'date_modified': {
+                            'gte': '{}'.format(min_date.isoformat()),
+                            'lte': '{}'.format(max_date.isoformat())
+                        }
+                    }
+                }
+            }
+        )['count']
+        return (partial_database_count == partial_es_count, partial_database_count, partial_es_count)
+
+    def get_date_range_parts(self, min_date, max_date):
+        middle_date = min_date.average(max_date)
+        return {
+            'first_half': {'min_date': min_date, 'max_date': middle_date},
+            'second_half': {'min_date': middle_date, 'max_date': max_date}
+        }
+
+    def pseudo_bisection_method(self, min_date, max_date):
+        '''
+        Checks if the database count matches the ES count
+        If counts differ, split the date range in half and check counts in smaller ranges
+        Pseudo binary because it can't throw away half the results based on the middle value
+        '''
+        from bots.elasticsearch.bot import ElasticSearchBot
+
+        counts_match, db_count, es_count = self.check_counts_in_range(min_date, max_date)
+
+        if counts_match:
+            return
+        if db_count <= 500:
+            logger.info('Counts for {} to {} do not match. {} creativeworks in ES, {} creativeworks in database.'.format(min_date.format('%B %-d, %Y %I:%M:%S %p'), max_date.format('%B %-d, %Y %I:%M:%S %p'), es_count, db_count))
```

Comment:
Please use `%` formats for logging messages. It saves a bit of work if the logs aren't being rendered somewhere and will catch any formatting errors if there happen to be any.